js/signal_protocol_store.js

```diff
@@ -331,10 +331,10 @@
                     var promises = [];
                     while (sessions.length > 0) {
                         promises.push(new Promise(function(res) {
-                            sessions.pop().destroy().then(res);
+                            sessions.pop().destroy().then(res, res);
                         }));
                     }
-                    Promise.all(promises).then(resolve);
+                    Promise.all(promises).then(resolve, resolve);
```

Comment:
Yeah, but removeIdentityKey is only called in tests at this point. Now that we can overwrite keys when they changed there is never a need to remove them outside tests.